### PR TITLE
[SIMT] Add any_sync warp intrinsics

### PR DESCRIPTION
**Pull request by varinic**
Related issue = #4631 
Add any_sync intrinsics for cuda backend.

### CODE REVIEW

File: taichi/runtime/llvm/runtime.cpp [hunk 1]
```diff
@@ -1036,6 +1036,18 @@ f32 cuda_shfl_up_sync_f32(u32 mask, f32 val, i32 delta, int width) {
   return 0;
 }
 
+bool cuda_any_sync(u32 mask, bool bit) {
+  return false;
+}
+
+int32 cuda_any_i32(int32 predicate) {
+  return (int32)cuda_any_sync(UINT32_MAX, (bool)predicate);
```
Comment (YuCrazing) [inline taichi/runtime/llvm/runtime.cpp:1044]: I am not sure if these functions should return 0 or something else, we need @strongoier  and @qiao-bo to have a look.

Comment (qiao-bo) [reply taichi/runtime/llvm/runtime.cpp:1044]: This is fine here. Since only `cuda_any_sync()` is patched later in llvm_context.cpp

**Comment by qiao-bo** [comment]
Hi, can you please fix the conflict first? thanks ;/

File: taichi/runtime/llvm/runtime.cpp [hunk 2]
```diff
@@ -1036,6 +1036,18 @@ f32 cuda_shfl_up_sync_f32(u32 mask, f32 val, i32 delta, int width) {
   return 0;
 }
 
+bool cuda_any_sync(u32 mask, bool bit) {
+  return false;
+}
+
+int32 cuda_any_i32(int32 predicate) {
```
Comment (YuCrazing) [inline taichi/runtime/llvm/runtime.cpp:1043]: Is this function redundant? It seems that `cuda_any_i32` can be replaced by `cuda_any_sync_i32`. If you want to keep this function, please expose it to python and add corresponding unit tests. :)

**Review by YuCrazing — CHANGES_REQUESTED** [review]

File: taichi/runtime/llvm/runtime.cpp (hunk 2 above)
Comment (varinic) [reply taichi/runtime/llvm/runtime.cpp]: I will remove `cuda_any_i32`.

**Review by YuCrazing — APPROVED** [review]
LGTM!Thanks!

**Pull request merged**